src/js/validator.js

```diff
@@ -74,17 +74,18 @@ class Validator
 		if (this.isNotRequired(methods, value)) {
 			return;
 		}
-
+		let index = 0;
 		for (let method in methods) {
 			//if the rule required exits and there was an error, the
 			//stack errors method is called to keep track of rules
 			//that did not pass the validation.
-			if (this[methods[method]] && ! this[methods[method]](value)) {
+			if (this[methods[method]] && ! this[methods[method]](value, methods[index+1])) {
```

Comment:
If pattern matches, evaluate the min and max. I am open for Ideas, so we can pull this off!

Thanks